modules/sdk/project-templates/project-templates-rest-builder/src/main/java/com/liferay/project/templates/rest/builder/internal/RestBuilderProjectTemplateCustomizer.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.project.templates.rest.builder.internal;
+
+import com.liferay.project.templates.extensions.ProjectTemplateCustomizer;
+import com.liferay.project.templates.extensions.ProjectTemplatesArgs;
+import com.liferay.project.templates.extensions.util.WorkspaceUtil;
+
+import java.io.File;
+
+import java.nio.file.Path;
+
+import java.util.Properties;
+
+import org.apache.maven.archetype.ArchetypeGenerationRequest;
+import org.apache.maven.archetype.ArchetypeGenerationResult;
+
+/**
+ * @author Javier de Arcos
+ */
+public class RestBuilderProjectTemplateCustomizer
+	implements ProjectTemplateCustomizer {
+
+	@Override
+	public String getTemplateName() {
+		return "rest-builder";
+	}
+
+	@Override
+	public void onAfterGenerateProject(
+		ProjectTemplatesArgs projectTemplatesArgs, File destinationDir,
+		ArchetypeGenerationResult archetypeGenerationResult) {
+	}
+
+	@Override
+	public void onBeforeGenerateProject(
+			ProjectTemplatesArgs projectTemplatesArgs,
+			ArchetypeGenerationRequest archetypeGenerationRequest)
+		throws Exception {
+
+		String artifactId = archetypeGenerationRequest.getArtifactId();
+
+		String apiPath = ":" + artifactId + "-api";
+
+		String clientPath = ":" + artifactId + "-client";
+
+		File destinationDir = new File(
+			archetypeGenerationRequest.getOutputDirectory());
+
+		File workspaceDir = WorkspaceUtil.getWorkspaceDir(destinationDir);
+
+		if (workspaceDir != null) {
+			Path destinationDirPath = destinationDir.toPath();
+			Path workspaceDirPath = workspaceDir.toPath();
+
+			destinationDirPath = destinationDirPath.toAbsolutePath();
+			workspaceDirPath = workspaceDirPath.toAbsolutePath();
+
+			String relativePath = String.valueOf(
+				workspaceDirPath.relativize(destinationDirPath));
+
+			relativePath = relativePath.replace(File.separatorChar, ':');
+
+			apiPath = ":" + relativePath + ":" + artifactId + apiPath;
```

Comment:
I though SF would complain here when concatenating more than 3 strings

Reply:
I run SF several times and fix some issues but it does not complain about this one.

I'll give it a look to change it anyway

Reply:
When I check it it came to my mind again.

I thought about using StringBundler but it didn't feel right to add a dependency to petra inside a tool module that should be as free as dependencies as possible.

Checking now, the modules/sdk folder (parent of project-templates) has source-formatter exclusion files, so, in this case I think it is better leaving it like that